components/admin_console/brand_image_setting/brand_image_setting.jsx

```diff
@@ -62,86 +70,129 @@ export default class BrandImageSetting extends React.PureComponent {
 
             reader.readAsDataURL(this.state.brandImage);
         }
+
+        if (this.props.saving && this.state.brandImage && !this.props.disabled) {
+            this.handleImageSubmit();
+        }
+
+        if (this.props.saving && this.state.deleteBrandImage && !this.props.disabled) {
+            this.handleImageDelete();
+        }
     }
 
     handleImageChange() {
         const element = $(this.refs.fileInput);
-
         if (element.prop('files').length > 0) {
+            this.props.onChange('saveNeeded', 'config');
             this.setState({
                 brandImage: element.prop('files')[0],
-                status: UploadStatuses.DEFAULT,
             });
         }
     }
 
-    handleImageSubmit(e) {
-        e.preventDefault();
+    handleImageDelete() {
+        this.setState({
+            error: '',
+        });
+        deleteBrandImage(
+            () => {
+                this.setState({
+                    deleteBrandImage: false,
+                    brandImageExists: false,
+                    brandImage: null,
+                });
+            },
+            (err) => {
+                this.setState({
+                    error: err.message,
+                });
+                this.props.onChange('saveNeeded', 'config');
+            }
+        );
+    }
 
+    handleImageSubmit() {
         if (!this.state.brandImage) {
             return;
         }
-
-        if (this.state.status === UploadStatuses.LOADING) {
-            return;
-        }
-
         this.setState({
             error: '',
-            status: UploadStatuses.LOADING,
         });
-
         uploadBrandImage(
             this.state.brandImage,
             () => {
                 this.setState({
                     brandImageExists: true,
                     brandImage: null,
                     brandImageTimestamp: Date.now(),
-                    status: UploadStatuses.COMPLETE,
                 });
             },
             (err) => {
                 this.setState({
                     error: err.message,
-                    status: UploadStatuses.DEFAULT,
                 });
+                this.props.onChange('saveNeeded', 'config');
             }
         );
     }
 
     render() {
-        let btnPrimaryClass = 'btn';
-        if (this.state.brandImage) {
-            btnPrimaryClass += ' btn-primary';
-        }
-
         let letbtnDefaultClass = 'btn';
         if (!this.props.disabled) {
             letbtnDefaultClass += ' btn-default';
         }
 
         let img = null;
+        let displayImgOverlay = false;
+        const IMGOVERLAY = (
```

Comment:
Can you use camel case for this name since it's not a constant?

Reply:
Resolved!